credoai/evaluators/identity_verification.py

```diff
@@ -0,0 +1,314 @@
+            target-subject-id: unique identifier of the target subject
+            target-subject-data-sample: data sample from the target subject
+    subjects_sensitive_features : pd.DataFrame of shape (n_subjects, n_sensitive_feature_names), optional
+        Sensitive features of all subjects present in pairs dataframe
+        If provided, disaggregated performance assessment is also performed.
+        This can be the columns you want to perform segmentation analysis on, or
+        a feature related to fairness like 'race' or 'gender'
+        Required columns:
+            subject-id: id of subjects. Must cover all the subjects inlcluded in `pairs` dataframe
+            other columns with arbitrary names for sensitive features
+    similarity_thresholds : list
+        list of similarity score thresholds
+        Similarity equal or greater than a similarity score threshold means match
+    comparison_levels : list
+        list of comparison levels. Options:
+            sample: it means a match is observed for every sample pair. Sample-level comparison represent
+                a use case where only two samples (such as a real time selfie and stored ID image) are
+                used to confirm an identity.
+            subject: it means if any pairs of samples for the same subject are a match, the subject pair
+                is marked as a match. Some identity verification use cases improve overall accuracy by storing
+                multiple samples per identity. Subject-level comparison mirrors this behavior.
+    """
+
+    def __init__(
+        self,
+        similarity_thresholds: list = [90, 95],
+        comparison_levels: list = ["sample", "subject"],
+    ):
+        self.similarity_thresholds = similarity_thresholds
+        self.comparison_levels = comparison_levels
+
+    name = "IdentityVerification"
+    required_artifacts = ["model", "assessment_data"]
+
+    def _setup(self):
+        self.pairs = self.assessment_data.pairs
+        try:
+            self.subjects_sensitive_features = (
+                self.assessment_data.subjects_sensitive_features
+            )
+            sensitive_features_names = list(self.subjects_sensitive_features.columns)
+            sensitive_features_names.remove("subject-id")
+            self.sensitive_features_names = sensitive_features_names
+        except:
+            self.subjects_sensitive_features = None
+
+        self.pairs["similarity_score"] = self.model.compare(
+            [
+                list(pair)
+                for pair in zip(
+                    self.pairs["source-subject-data-sample"].tolist(),
+                    self.pairs["target-subject-data-sample"].tolist(),
+                )
+            ]
+        )
+
+        self.pairs["match"] = self.pairs.apply(
+            lambda x: 1 if x["source-subject-id"] == x["target-subject-id"] else 0,
+            axis=1,
+        )
+
+        self.results = list()
```

Comment:
If you initiate the base class (see my comment on the init method above), this line becomes unnecessary. 
Please remove after the update to the init method :)